src/Board.hs

```diff
@@ -60,14 +62,14 @@ scrambleBoard board@(Board (Dimensions width height) _) startingBoard gen =
                in verifyBoard newBoard gen iterations
           verifyBoard board gen iterations
               | board == startingBoard = scrambleBoardLoop board gen iterations
-              | otherwise              = scrambleBoardLoop board gen (pred iterations)
+              | otherwise              = scrambleBoardLoop board gen (iterations - 1)
 
 -- | Randomly generate a new board, given dimensions.
 newBoard :: Dimensions -> IO Board
 newBoard d = do
     newGen <- R.newStdGen
-    return $ scrambleBoard startingBoard startingBoard newGen
-    where startingBoard = calculateStartingBoard $ Board d []
+    return (scrambleBoard startingBoard startingBoard newGen)
+    where startingBoard = calculateStartingBoard (Board d [])
```

Comment:
Just noticed that [Chris Done's style guide](https://github.com/chrisdone/haskell-style-guide) also suggests against the `$` operator. I, too, used to use lots of `$`s, and I was outraged when I saw Chris Done and Gabriel Gonzales discourage the use of `$`, but I have become used to it (and agree with it) now.